bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusBaseHandler.java

```diff
@@ -142,14 +152,23 @@ protected Location getLocation() {
     protected @Nullable T getAppliance(OndusService ondusService) {
         try {
             BaseAppliance appliance = ondusService.getAppliance(getRoom(), config.applianceId).orElse(null);
-            if (appliance.getType() != getType()) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "Thing is not a GROHE SENSE Guard device.");
-                return null;
+            if (appliance != null) {
+                if (appliance.getType() != getType()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Thing is not a GROHE SENSE Guard device.");
```

Comment:
Consider moving text to src/main/resources/OH-INF/i18n/groheondus.properties to allow for localization.